src/dmd/e2ir.d

```diff
@@ -5678,6 +5678,18 @@ private elem *toElemStructLit(StructLiteralExp sle, IRState *irs, TOK op, Symbol
         e = el_combine(e, e1);
     }
 
+    if (global.params.useInvariants && sle.sd.inv)
```

Comment:
Inserting the invariant call here seems strange. Everywhere else it is controlled by `addPostInvariant`